### PR TITLE
(BIDS-1454) Add location identifier to handleTemplateError

### PR DESCRIPTION
**Pull request by recy21**

### CODE REVIEW

File: handlers/epochs.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func Epochs(w http.ResponseWriter, r *http.Request) {
 		DisplayStart:    start,
 	}
 
-	if handleTemplateError(w, r, epochsTemplate.ExecuteTemplate(w, "layout", data)) != nil {
+	if handleTemplateError(w, r, "epochs.go / Done", epochsTemplate.ExecuteTemplate(w, "layout", data)) != nil {
```
Comment (D13ce) [inline handlers/epochs.go:121]: This looks like a copy and paste error and should be
`if handleTemplateError(w, r, "epochs.go / Epochs", epochsTemplate.ExecuteTemplate(w, "layout", data)) != nil {`
to be consistent with your other changes.

File: handlers/eth1Token.go [hunk 2]
```diff
@@ -118,7 +119,7 @@ func Eth1Token(w http.ResponseWriter, r *http.Request) {
 		Price:            template.HTML(fmt.Sprintf("<span>$%s</span><span>@ %.6f</span>", string(metadata.Price), ethExchangeRate)),
 	}
 
-	if handleTemplateError(w, r, eth1TokenTemplate.ExecuteTemplate(w, "layout", data)) != nil {
+	if handleTemplateError(w, r, "eth1Token.go / Eth1Token", eth1TokenTemplate.ExecuteTemplate(w, "layout", data)) != nil {
```
Comment (D13ce) [inline handlers/eth1Token.go:122]: This looks like a copy and paste error and should be
if handleTemplateError(w, r, "eth1Token.go / Eth1Token / Done", ...
to be consistent with your other changes.

File: handlers/ethstore.go [hunk 3]
```diff
@@ -15,11 +15,7 @@ func EthStore(w http.ResponseWriter, r *http.Request) {
 	data := InitPageData(w, r, "services", "/ethstore", "ETH.STORE Statistics")
 	data.Data = services.LatestEthStoreStatistics()
 
-	err := ethStoreTemplate.ExecuteTemplate(w, "layout", data)
-
-	if err != nil {
-		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
-		http.Error(w, "Internal server error", http.StatusInternalServerError)
-		return
+	if handleTemplateError(w, r, "eth1Token.go / EthStore", ethStoreTemplate.ExecuteTemplate(w, "layout", data)) != nil {
```
Comment (D13ce) [inline handlers/ethstore.go:18]: This looks like a copy and paste error and should be
`if handleTemplateError(w, r, "ethstore.go / EthStore", ...`
to be correct.

File: handlers/pools.go [hunk 4]
```diff
@@ -47,7 +49,7 @@ func Pools(w http.ResponseWriter, r *http.Request) {
 
 	data.Data = poolData
 
-	if handleTemplateError(w, r, poolsServicesTemplate.ExecuteTemplate(w, "layout", data)) != nil {
+	if handleTemplateError(w, r, "pools.go / Pools", poolsServicesTemplate.ExecuteTemplate(w, "layout", data)) != nil {
```
Comment (D13ce) [inline handlers/pools.go:52]: This should be changed to
`if handleTemplateError(w, r, "pools.go / Pools / Done", ...`
to be consistent with your other changes.

File: handlers/block.go [hunk 5]
```diff
@@ -117,13 +117,14 @@ func Block(w http.ResponseWriter, r *http.Request) {
...
+		if handleTemplateError(w, r, "block.go / Block / ErrNoRows", blockFutureTemplate.ExecuteTemplate(w, "layout", data)) != nil {
 			return // an error has occurred and was processed
 		}
 		return
 	} else if err != nil {
-		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
-		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		if handleTemplateError(w, r, "block.go / Block / GetSlotPageData", err) != nil {
+			return // an error has occurred and was processed
+		}
```
Comment (D13ce) [inline handlers/block.go:127]: The http error code returned has changed from `http.StatusInternalServerError` to `http.StatusServiceUnavailable`.
Is this change intentional?

File: handlers/epoch.go [hunk 6]
```diff
@@ -171,7 +171,7 @@ func Epoch(w http.ResponseWriter, r *http.Request) {
 		err = epochTemplate.ExecuteTemplate(w, "layout", data)
 	}
 
-	if handleTemplateError(w, r, err) != nil {
+	if handleTemplateError(w, r, "epoch.go / Epoch / Done", err) != nil {
 		return // an error has occurred and was processed
 	}
```
Comment (D13ce) [inline handlers/epoch.go:176]: Minor: You could replace all this by just `handleTemplateError(w, r, "epoch.go / Epoch / Done", err)` and get the same result.

File: handlers/eth1Account.go [hunk 7]
```diff
@@ -121,8 +121,9 @@ func Eth1Address(w http.ResponseWriter, r *http.Request) {
 	// }
 
 	if err := g.Wait(); err != nil {
-		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
-		http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+		if handleTemplateError(w, r, "eth1Account.go / Block / g.Wait()", err) != nil {
+			return // an error has occurred and was processed
+		}
 		return
```
Comment (D13ce) [inline handlers/eth1Account.go:127]: Minor:
```
if x {
	return
}
return
```
could be shortened to
```
x
return
```

File: handlers/eth1Token.go [hunk 8]
```diff
@@ -61,8 +61,9 @@ func Eth1Token(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err := g.Wait(); err != nil {
-		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
-		http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+		if handleTemplateError(w, r, "eth1Token.go / Eth1Token / g.Wait()", err) != nil {
+			return // an error has occurred and was processed
+		}
 		return
```
Comment (D13ce) [inline handlers/eth1Token.go:67]: Minor:
```
if x {
	return
}
return
```
could be shortened to
```
x
return
```

And again, I feel like you are misusing this function as `err` is not caused by anything related to templates.

File: handlers/pools.go [hunk 9]
```diff
@@ -23,15 +23,17 @@ func Pools(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
 	performanceData, err := services.ChartHandlers["historic_pool_performance"].DataFunc()
 	if err != nil {
-		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
-		http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+		if handleTemplateError(w, r, "pools.go / Pools / historic_pool_performance", err) != nil {
+			return // an error has occurred and was processed
+		}
```
Comment (D13ce) [inline handlers/pools.go:36]: Minor:
```
if x {
	return
}
return
```
could be shortened to
```
x
return
```

And again, I feel like you are misusing this function as `err` is not caused by anything related to templates.

File: handlers/user.go [hunk 10]
```diff
@@ -212,7 +212,7 @@ func UserAuthorizeConfirm(w http.ResponseWriter, r *http.Request) {
 
 	err = authorizeTemplate.ExecuteTemplate(w, "layout", data)
 	if err != nil {
-		logger.Errorf("error executing template for %v route: %v", r.URL.String(), err)
+		logger.Errorf("error executing template [user.go / UserAuthorizeConfirm] for %v route: %v", r.URL.String(), err)
```
Comment (D13ce) [inline handlers/user.go:215]: Should print "()" instead of "[]".

**Review by D13ce — CHANGES_REQUESTED** [review]
A few minor things I have addressed via comments, some of them (wrong log messages) require to be fixed as this is about log messages improvements.

I have also left some general suggestions.

File: handlers/user.go (hunk 10 above)
Comment (recy21) [reply handlers/user.go:215]: It's on purpose, as it's not using handleTemplateError
